core/ledger/kvledger/hashcheck_pvtdata.go

```diff
@@ -10,27 +10,47 @@ import (
 	"bytes"
 
 	"github.com/hyperledger/fabric-protos-go/ledger/rwset"
+	"github.com/hyperledger/fabric-protos-go/ledger/rwset/kvrwset"
 	"github.com/hyperledger/fabric/common/ledger/blkstorage"
 	"github.com/hyperledger/fabric/common/util"
 	"github.com/hyperledger/fabric/core/ledger"
 	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/rwsetutil"
+	"github.com/hyperledger/fabric/core/ledger/pvtdatastorage"
 	"github.com/hyperledger/fabric/protoutil"
 )
 
 // constructValidAndInvalidPvtData computes the valid pvt data and hash mismatch list
-// from a received pvt data list of old blocks.
-func constructValidAndInvalidPvtData(reconciledPvtdata []*ledger.ReconciledPvtdata, blockStore *blkstorage.BlockStore) (
+// from a received pvt data list of old blocks. The reconciled data elements that belong
+// to a block which is not available in the blockstore (i.e., the block is prior to or equal
+// to the lastBlockInBootSnapshot), the boot KV hashes in the private data store are used for
+// verifying the hashes. Further, in this case, the write-set of a collection is trimmed in order
+// to remove the key-values that were not present in the snapshot (most likely because, they
+//	were over-written in a later version)
```

Comment:
Not related to this PR. Given this point and also we don't support history queries on private data, we can optimize the reconciler to ignore the old version of a missing data.